kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/config/CredentialsValidator.java

```diff
@@ -46,6 +47,9 @@ public CredentialsValidator() {
       PROJECT_CONFIG, KEY_SOURCE_CONFIG
   ));
 
+  public static final String ERROR_MESSAGE_KEY_SHOULD_NOT_BE_PROVIDED_IF_ADC = KEYFILE_CONFIG + " should not be "
+          + "provided if " + KEY_SOURCE_CONFIG + " is " + KeySource.APPLICATION_DEFAULT;
```

Comment:
This style (declaring static error strings at the top of a class file) isn't used anywhere else in the code base AFAIK, and I think we should keep it that way. It adds distance between where these error messages are declared and where they are used, it quickly becomes unreadable if/when a dynamic message needs to be constructed (for example, when referencing user input in the error message like a path to a file), and it can quickly clutter up the top of source files where other important static constants are declared and can become harder to find.

I see the motivation is to ensure that the correct error message is rendered to the user in the `CredentialsValidatorTest::testKeyShouldNotBeProvidedIfUsingApplicationDefaultCredentials` test case, which is understandable since it automatically updates that test case for us if the wording in the error message ever changes. But given that the error message isn't likely to be updated very frequently once this PR is merged and, if it is, the failing test can be fixed with negligible effort, I think it may be best to just hardcode the testing logic to use the error string. Or, if that seems too brittle to you, the test can be changed to use a substring match (i.e., `String::contains`).

Reply:
Ok, I see your point. Done :)